### PR TITLE
[Action Cable] Add `WebSocket` and `logger` configuration options

### PR DESCRIPTION
**Pull request by maclover7**
[Javan Makhmali, Jon Moss]

r? @javan

### CODE REVIEW

File: actioncable/test/javascript/src/unit/action_cable_test.coffee [hunk 1]
```diff
@@ -22,3 +22,19 @@ module "ActionCable", ->
...
+      ActionCable.WebSocket = ''
+      assert.equal ActionCable.WebSocket, ''
+
+  module 'logger', ->
+    test 'default is window.console', (assert) ->
+      assert.equal ActionCable.logger, window.console
+
+    test 'configurable', (assert) ->
+      ActionCable.logger = ''
+      assert.equal ActionCable.logger, ''
```
Comment (javan) [inline actioncable/test/javascript/src/unit/action_cable_test.coffee]: How about grouping all of these new tests in one `module "Adapters"`? And, I know this is nit picky, but let's use double quotes everywhere to match the existing style.

File: actioncable/test/javascript/src/test_helpers/mock_websocket.coffee [hunk 2]
```diff
@@ -18,4 +16,3 @@ QUnit.testDone ->
   if server?
     server.clients().forEach (client) -> client.close()
     server.close()
-    window.WebSocket = NativeWebSocket
```
Comment (javan) [inline actioncable/test/javascript/src/test_helpers/mock_websocket.coffee]: I think we should drop this file entirely swap out `WebSocket` as needed.

```coffee
module "<Any group of tests that use WebSocket>", (hooks) ->
  originalWebSocket = ActionCable.WebSocket
  
  hooks.beforeEach ->
    ActionCable.WebSocket = MockWebSocket

  hooks.afterEach ->
    ActionCable.WebSocket = originalWebSocket
```

Comment (javan) [inline actioncable/test/javascript/src/test_helpers/mock_websocket.coffee]: ^ Not a complete replacement there since it doesn't incorporate `MockServer`.

Comment (maclover7) [inline actioncable/test/javascript/src/test_helpers/mock_websocket.coffee]: 👍 how does eeed98c1d586ae13688392d1808d1fa38c1887cf look?

Comment (javan) [inline actioncable/test/javascript/src/test_helpers/mock_websocket.coffee]: Wrapped that up even more https://github.com/maclover7/rails/pull/3

**Comment by maclover7** [comment]
👍

**Pull request merged**